In [3]:
# Day 1: Data Overview and Project Setup

# Step 1: Import Libraries
import pandas as pd

# Step 2: Load the dataset
df = pd.read_csv('/content/drive/MyDrive/DataSet/DataScientist.csv')  # Adjust path if needed

# Step 3: Display first 5 rows
df.head()


Unnamed: 0  index                          Job Title  \
0           0      0              Senior Data Scientist   
1           1      1  Data Scientist, Product Analytics   
2           2      2               Data Science Manager   
3           3      3                       Data Analyst   
4           4      4             Director, Data Science   

                Salary Estimate  \
0  $111K-$181K (Glassdoor est.)   
1  $111K-$181K (Glassdoor est.)   
2  $111K-$181K (Glassdoor est.)   
3  $111K-$181K (Glassdoor est.)   
4  $111K-$181K (Glassdoor est.)   

                                     Job Description  Rating  \
0  ABOUT HOPPER\n\nAt Hopper, we’re on a mission ...     3.5   
1  At Noom, we use scientifically proven methods ...     4.5   
2  Decode_M\n\nhttps://www.decode-m.com/\n\nData ...    -1.0   
3  Sapphire Digital seeks a dynamic and driven mi...     3.4   
4  Director, Data Science - (200537)\nDescription...     3.4   

                      Company Name       Location      Headquarters  \
0                      Hopper\n3.5   New York, NY  Montreal, Canada   
1                     Noom US\n4.5   New York, NY      New York, NY   
2                         Decode_M   New York, NY      New York, NY   
3            Sapphire Digital\n3.4  Lyndhurst, NJ     Lyndhurst, NJ   
4  United Entertainment Group\n3.4   New York, NY      New York, NY   

                     Size  Founded  Type of ownership  \
0   501 to 1000 employees     2007  Company - Private   
1  1001 to 5000 employees     2008  Company - Private   
2       1 to 50 employees       -1            Unknown   
3    201 to 500 employees     2019  Company - Private   
4     51 to 200 employees     2007  Company - Private   

                    Industry                  Sector  \
0            Travel Agencies        Travel & Tourism   
1  Health, Beauty, & Fitness       Consumer Services   
2                         -1                      -1   
3                   Internet  Information Technology   
4    Advertising & Marketing       Business Services   

                    Revenue               Competitors Easy Apply  
0  Unknown / Non-Applicable                        -1         -1  
1  Unknown / Non-Applicable                        -1         -1  
2  Unknown / Non-Applicable                        -1       True  
3  Unknown / Non-Applicable      Zocdoc, Healthgrades         -1  
4  Unknown / Non-Applicable  BBDO, Grey Group, Droga5         -1

Basic Exploration

In [4]:
# Check shape of dataset (rows, columns)
print("Dataset Shape:", df.shape)

# Display column names
print("\nColumn Names:\n", df.columns.tolist())

# Check for missing values
print("\nMissing Values:\n", df.isnull().sum())

# Check data types
print("\nData Types:\n", df.dtypes)

# Check for quality
print("\nQuality Check:\n", df.describe())

Dataset Shape: (3909, 17)

Column Names:
 ['Unnamed: 0', 'index', 'Job Title', 'Salary Estimate', 'Job Description', 'Rating', 'Company Name', 'Location', 'Headquarters', 'Size', 'Founded', 'Type of ownership', 'Industry', 'Sector', 'Revenue', 'Competitors', 'Easy Apply']

Missing Values:
 Unnamed: 0           0
index                0
Job Title            0
Salary Estimate      0
Job Description      0
Rating               0
Company Name         0
Location             0
Headquarters         0
Size                 0
Founded              0
Type of ownership    0
Industry             0
Sector               0
Revenue              0
Competitors          0
Easy Apply           0
dtype: int64

Data Types:
 Unnamed: 0             int64
index                  int64
Job Title             object
Salary Estimate       object
Job Description       object
Rating               float64
Company Name          object
Location              object
Headquarters          object
Size                  object
F

## Day 1: Data Overview & Quality Check Summary  
**Dataset:** `DataScientist.csv`

---

### Objective:
To load the dataset, understand its structure, and assess its quality through missing value analysis, data types, and consistency checks.

---

### Exploratory Steps Performed:

1. **Dataset Shape:**  
   - Rows: `3,909`  
   - Columns: `17`

2. **Missing Values:**  
   -  No missing values in any column

3. **Duplicate Rows:**  
   -  No duplicate entries found

4. **Data Types & Structure:**  
   - `Salary Estimate` is a **text column** and needs conversion for numeric analysis  
   - All other fields have appropriate types (e.g., `Rating` as float, `Founded` as int)

5. **Column Quality Review:**  
   - `Job Title`, `Company Name`, and similar fields contain **inconsistent casing and whitespace** — flagged for cleaning  
   - `Job Description` entries are **detailed and unique** — good for text mining / NLP  
   - `Unnamed: 0` and `index` columns are **redundant row identifiers** — marked for removal

---